### PR TITLE
refactor(window.c): reduce scope of locals

### PR DESCRIPTION
**Pull request by lewis6991**

### CODE REVIEW

File: src/nvim/window.c [hunk 1]
```diff
@@ -490,7 +487,8 @@ void do_window(int nchar, long Prenum, int xchar)
 wingotofile:
     CHECK_CMDWIN;
 
-    ptr = (char *)grab_file_name(Prenum1, &lnum);
+    linenr_T lnum = -1;
+    char *ptr = (char *)grab_file_name(Prenum1, &lnum);
```
Comment (zeertzjq) [inline src/nvim/window.c:491]: Moving the declaration of `ptr` here looks a bit strange.

File: src/nvim/window.c [hunk 2]
```diff
@@ -1064,30 +1054,33 @@ int win_split_ins(int size, int flags, win_T *new_wp, int dir)
...
-    int minwidth;
-
-    layout = FR_ROW;
+  bool do_equal = false;
+  int oldwin_height = 0;
+  const int layout = flags & WSP_VERT ? FR_ROW : FR_COL;
+  bool did_set_fraction = false;
 
+  if (flags & WSP_VERT) {
+    int available;
```
Comment (zeertzjq) [inline src/nvim/window.c:1063]: `available` can be moved after `minwidth`

File: src/nvim/window.c [hunk 3]
```diff
@@ -1970,6 +1963,8 @@ void win_move_after(win_T *win1, win_T *win2)
       return;
     }
 
+    int height;
```
Comment (zeertzjq) [inline src/nvim/window.c:1966]: The scope of `height` can be smaller, in the two branches.

File: src/nvim/window.c [hunk 4]
```diff
@@ -2193,9 +2187,10 @@ static void win_equal_rec(win_T *next_curwin, bool current, frame_T *topfr, int
...
         // Compute the maximum number of windows horiz. in "fr".
-        n = frame_minwidth(fr, NOWIN);
+        int n = frame_minwidth(fr, NOWIN);
         wincount = (n + (fr->fr_next == NULL ? extra_sep : 0)) / ((int)p_wmw + 1);
-        m = frame_minwidth(fr, next_curwin);
+        int m = frame_minwidth(fr, next_curwin);
+        bool hnc;
         if (has_next_curwin) {
           hnc = frame_has_win(fr, next_curwin);
         } else {
```
Comment (zeertzjq) [inline src/nvim/window.c:2196]: This can be a ternary

File: src/nvim/window.c [hunk 5]
```diff
@@ -2321,9 +2319,10 @@ static void win_equal_rec(win_T *next_curwin, bool current, frame_T *topfr, int
...
         // Compute the maximum number of windows vert. in "fr".
-        n = frame_minheight(fr, NOWIN);
+        int n = frame_minheight(fr, NOWIN);
         wincount = get_maximum_wincount(fr, (n + (fr->fr_next == NULL ? extra_sep : 0)));
-        m = frame_minheight(fr, next_curwin);
+        int m = frame_minheight(fr, next_curwin);
+        bool hnc;
         if (has_next_curwin) {
           hnc = frame_has_win(fr, next_curwin);
         } else {
```
Comment (zeertzjq) [inline src/nvim/window.c:2328]: This can be a ternary

File: src/nvim/window.c (hunk 1 above)
Comment (lewis6991) [reply src/nvim/window.c:491]: Didn't realize this was in scope for other cases.

I've surrounded in `{}` and added `ptr` decls for the other cases.

File: src/nvim/window.c (hunk 4 above)
Comment (lewis6991) [reply src/nvim/window.c:2196]: Or even better:

```c
        bool hnc = has_next_curwin && frame_has_win(fr, next_curwin);
```

File: src/nvim/window.c [hunk 6]
```diff
@@ -4644,20 +4610,18 @@ static void win_goto_ver(bool up, long count)
...
-  foundfr = wp->w_frame;
+  frame_T *foundfr = wp->w_frame;
 
   if (wp->w_floating) {
     return win_valid(prevwin) && !prevwin->w_floating ? prevwin : firstwin;
   }
 
   while (count--) {
+    frame_T *nfr;
+
```
Comment (zeertzjq) [inline src/nvim/window.c:4621]: Removing this blank line looks more consistent with code above

**Review by zeertzjq — APPROVED** [review]

File: src/nvim/window.c (hunk 6 above)
Comment (lewis6991) [reply src/nvim/window.c:4621]: I have a tendency to insert linebreaks if the line below is a comment to enforce what statement the comment is associated with.

```C
int a;
int b;
```

```C
int a;

// comment
int b;
```

but for this case I'll just remove.

**Pull request merged**